packages/protocol/test-sol/common/FeeCurrencyDirectory.t.sol

```diff
@@ -21,6 +24,23 @@ contract FeeCurrencyDirectoryTestBase is Test {
   }
 }
 
+contract TestE2ESetCurrencyConfig is Test, Devchain {
```

Comment:
I'd love some advice on naming conventions for file names, contract names and test names.

At the moment, I added a new contract ("`TestE2ESetCurrencyConfig`") and test ("`test_ShouldAllowOwnerSetCurrencyConfig`") to the existing `FeeCurrencyDirectory.t.sol` file. 

https://github.com/celo-org/celo-monorepo/blob/0ca1c67848924fb2b5458c83c9d4a9c498939cb4/packages/protocol/test-sol/common/FeeCurrencyDirectory.t.sol#L27-L29

We'll want to call `forge test` with some argument that separates unit tests from e2e tests, otherwise the tests will fail.

https://github.com/celo-org/celo-monorepo/blob/0ca1c67848924fb2b5458c83c9d4a9c498939cb4/packages/protocol/migrations_sol/run_e2e_tests_in_anvil.sh#L9-L14

Should we keep unit and e2e tests in the same file (`FeeCurrencyDirectory.t.sol`), and distinguish tests at the contract name ("`TestE2ESetCurrencyConfig`")? 

We can use a regex pattern to distinguish unit and e2e tests, if that's our preference:

> `--match-contract` _regex_ : Only run tests in contracts matching the specified regex pattern.

Source: [`forge test` docs](https://book.getfoundry.sh/reference/forge/forge-test#test-options)

Reply:
Personally I would prefer to have two separate files since the setup for tests is different.

Reply:
We could have a directory structure like 

```
test-sol/unit/
test-sol/e2e/
test-sol/integration/
```

That way we could run tests with something like this:

```
# unit tests
forge test --match-path "*test/unit/*"

# e2e tests
forge test --match-path "*test/e2e/*"

# integration tests
forge test --match-path "*test/integration/*"
```

Reply:
I've given this a go in https://github.com/celo-org/celo-monorepo/pull/11020/commits/1a66ba1f6ffb03365d408b5b80b668e13edc0695 and https://github.com/celo-org/celo-monorepo/pull/11020/commits/80ff4e2a58ec1ba1805896327f94b40523c2a052. 

Now the script looks like this

https://github.com/celo-org/celo-monorepo/blob/80ff4e2a58ec1ba1805896327f94b40523c2a052/packages/protocol/migrations_sol/run_e2e_tests_in_anvil.sh#L8-L13